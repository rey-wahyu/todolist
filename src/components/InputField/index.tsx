import { Button, InputBase } from "@mui/material";
import { makeStyles } from "@mui/styles";
import React, { useRef } from "react";

interface Props {
  todo: string;
  setTodo: React.Dispatch<React.SetStateAction<string>>;
  handleAdd: (e: React.FormEvent) => void;
}

const useStyles = makeStyles({
  input: {
    display: 'flex',
    width: '95%',
    position: 'relative',
    alignItems: 'center'
  },

  inputSubmit: {
    position: 'absolute',
    width: '50px',
    height: '50px',
    margin: '12px',
    borderRadius: '50px',
    right: '0px',
    border: 'none',
    fontSize: '15px',
    backgroundColor: '#2f74c0',
    color: 'white',
    transition: '0.2s all',
    boxShadow: '0 0 10px black',

    '&:hover': {
      backgroundColor: '#388ae2'
    },

    '&:active': {
      transform: 'scale(0.8);',
      boxShadow: '0 0 5px black'
    }
  },

  inputBox: {
    backgroundColor: 'white',
    color: 'black',
    width: '100%',
    borderRadius: '50px',
    padding: '15px 30px',
    fontSize: '25px',
    border: 'none',
    transition: '0.2s',
    boxShadow: 'inset 0 0 5px black',

    '&:focus': {
      boxShadow: '0 0 10px 1000px rgba(0, 0, 0, 0.5)',
      outline: 'none'
    }
  }
})

const InputField: React.FC<Props> = ({ todo, setTodo, handleAdd }) => {
  const inputRef = useRef<HTMLInputElement>(null);
  const classes = useStyles();

  return (
    <form
      className={classes.input} onSubmit={(e) => {
        handleAdd(e);
        inputRef.current?.blur();
      }}>
      <InputBase
        type="text"
        size="small"
        placeholder="Enter a Task"
        value={todo}
        ref={inputRef}
        onChange={(e) => setTodo(e.target.value)}
        className={classes.inputBox}
      />
      <Button type="submit" variant="text" className={classes.inputSubmit}>Go</Button>
    </form>
  );
};

export default InputField;
