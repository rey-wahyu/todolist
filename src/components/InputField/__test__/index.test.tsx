import { cleanup } from '@testing-library/react';
import InputField from "../index";
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

afterEach(cleanup)

const handleAdd = jest.fn(e => e.preventDefault())
const setTodo = jest.fn
const todo = ""

// jest.mock('../index')

const props = {
    handleAdd,
    setTodo,
    todo,
}

const wrapper = shallow(<InputField {...props} />)

describe("<InputField /> rendering", () => {
    test("Should be render in window", () => {
        expect(wrapper).toHaveLength(1)
    })

    test("Form field should be render", () => {
        const input = wrapper.find("form").debug()
        expect(input).toBeInTheDocument
    })

    test("Should 0 length value i input field after Go Button clicked", () => {
        wrapper.find("ForwardRef(Button)").simulate('click')
        const input = wrapper.find("ForwardRef(InputBase)")
        expect(input.text().length).toBe(0)
    })

    describe("Testing input", () => {
        test("Button should be disamble if Input with empty value", () => {
            expect(wrapper.find("ForwardRef(Button)")).toBeDisabled
        })
    
        test("Success add new todolist ", () => {
            const todo = "testing"
    
            wrapper.find("ForwardRef(InputBase)").simulate('change', { target: { value: todo}}) 
            wrapper.find("ForwardRef(Button)").simulate('click')
            expect(handleAdd).toHaveLength(1)
        })
    })

    test("Get a snapshop", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    })
})
