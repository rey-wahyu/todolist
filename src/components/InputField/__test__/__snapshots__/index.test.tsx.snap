// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> rendering Get a snapshop 1`] = `
"<form data-testid=\\"addTodo\\" className=\\"makeStyles-input-1\\" onSubmit={[Function: onSubmit]}>
  <ForwardRef(InputBase) type=\\"text\\" size=\\"small\\" placeholder=\\"Enter a Task\\" name=\\"todo\\" value=\\"\\" onChange={[Function: onChange]} className=\\"makeStyles-inputBox-3\\" />
  <ForwardRef(Button) type=\\"submit\\" variant=\\"text\\" className=\\"makeStyles-inputSubmit-2\\">
    Go
  </ForwardRef(Button)>
</form>"
`;
